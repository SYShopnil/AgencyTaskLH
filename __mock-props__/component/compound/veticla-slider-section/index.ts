import {
  IVerticalCarouselMain,
  IVerticalCarouselSectionMain,
} from '@src/types/compound/vertical-slider-section';

const slides: IVerticalCarouselMain[] = [
  {
    key: 0,
    content: {
      cardImg: '/assets/media/cardImageOne.png',
      cardNo: '01',
      desc: `Your data is organized, cleaned, and given context as we interpret
            it. Your data gains meaning through the semantic layer, making it
            simpler for business users to comprehend. We assist you in gaining a
            clear understanding of what it signifies so you can take action.`,
      heading: 'Pre-processing',
      isActive: true,
      slideCardDataTestId: 'slideCardTestIdOne',
    },
  },
  {
    key: 1,
    content: {
      cardImg: '/assets/media/cardImageTwo.png',
      cardNo: '02',
      desc: `Your data is organized, cleaned, and given context as we interpret
            it. Your data gains meaning through the semantic layer, making it
            simpler for business users to comprehend. We assist you in gaining a
            clear understanding of what it signifies so you can take action.`,
      heading: 'Pre-processing',
      isActive: true,
      slideCardDataTestId: 'slideCardTestIdOne',
    },
  },
  {
    key: 2,
    content: {
      cardImg: '/assets/media/cardImageThree.png',
      cardNo: '03',
      desc: `Your data is organized, cleaned, and given context as we interpret
            it. Your data gains meaning through the semantic layer, making it
            simpler for business users to comprehend. We assist you in gaining a
            clear understanding of what it signifies so you can take action.`,
      heading: 'Pre-processing',
      isActive: true,
      slideCardDataTestId: 'slideCardTestIdOne',
    },
  },
  {
    key: 3,
    content: {
      cardImg: '/assets/media/cardImageFour.png',
      cardNo: '04',
      desc: `Your data is organized, cleaned, and given context as we interpret
            it. Your data gains meaning through the semantic layer, making it
            simpler for business users to comprehend. We assist you in gaining a
            clear understanding of what it signifies so you can take action.`,
      heading: 'Pre-processing',
      isActive: true,
      slideCardDataTestId: 'slideCardTestIdOne',
    },
  },
  {
    key: 4,
    content: {
      cardImg: '/assets/media/cardImageFour.png',
      cardNo: '01',
      desc: `Your data is organized, cleaned, and given context as we interpret
            it. Your data gains meaning through the semantic layer, making it
            simpler for business users to comprehend. We assist you in gaining a
            clear understanding of what it signifies so you can take action.`,
      heading: 'Pre-processing',
      isActive: true,
      slideCardDataTestId: 'slideCardTestIdOne',
    },
  },
];

export const verticalSliderSectionMockData: IVerticalCarouselSectionMain = {
  slides,
};
