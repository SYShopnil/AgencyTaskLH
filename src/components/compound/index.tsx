export { HeroSection } from './hero-section';
export { HeroWithOnScrollRotation } from './hero-with-on-scroll-rotaion-section';
export { PartnerSection } from './partner-section';
