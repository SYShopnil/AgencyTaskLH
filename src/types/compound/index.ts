export type { IHeader } from './header';
export type { IHeroWithOnScrollRotation } from './hero-with-on-scroll-rotation-section';
export type { IPartnerSection } from './partner-section';
