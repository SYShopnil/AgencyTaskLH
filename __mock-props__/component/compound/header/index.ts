import { IHeader } from '@src/types/compound';
import { IconName } from '@src/types/root/_icon';
import { navBarMockData } from '../../root/navbar';

export const HeaderMockData: IHeader = {
  logo: IconName.Logo,
  navBarData: navBarMockData,
  dataTestId: 'headerCatchy',
};
