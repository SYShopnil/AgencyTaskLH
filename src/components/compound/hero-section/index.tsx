import { Button } from '@src/components/root';
import { BtnColorSchema } from '@src/types/root';
import React, { useState, useEffect } from 'react';
import ModuleStyle from './index.module.scss';

export const HeroSection = () => {
  const animationVariation = ['Primary', 'Secondary', 'Gradient'];
  const [activeAnimationVariationIndex, setActiveAnimationVariationIndex] =
    useState<number>(0);
  let activeButtonVariant: BtnColorSchema =
    BtnColorSchema.SolidBgPrimary500TextNeutral500;
  useEffect(() => {
    const animationLoopHandler = () => {
      setActiveAnimationVariationIndex(
        (prevIndex) => (prevIndex + 1) % animationVariation.length
      );
    };

    const interval = setInterval(animationLoopHandler, 1500);

    return () => clearInterval(interval);
  }, [animationVariation.length]);

  const activeAnimationVariation =
    animationVariation[activeAnimationVariationIndex];

  // handle the button animation part
  switch (true) {
    case activeAnimationVariation == animationVariation[0]: {
      activeButtonVariant = BtnColorSchema.SolidBgPrimary500TextNeutral500;
      break;
    }
    case activeAnimationVariation == animationVariation[1]: {
      activeButtonVariant = BtnColorSchema.SolidSecondary300TextNeutral500;
      break;
    }
    case activeAnimationVariation == animationVariation[2]: {
      activeButtonVariant = BtnColorSchema.SolidPrimaryGradientTextNeutral500;
      break;
    }
    default: {
      activeButtonVariant = BtnColorSchema.SolidBgPrimary500TextNeutral500;
    }
  }
  return (
    <div
      className={`flex justify-center items-center flex-col p-3`}
      data-testid="heroSectionCatcher"
    >
      <div className={`w-[100%] h-[auto] mb-[2.5rem]`}>
        <p className={`text-center text-3xl  text-[#E9EAEC]`}>
          <span
            className={` ${ModuleStyle['innovativeSolutionPart']} ${
              activeAnimationVariation == animationVariation[0]
                ? ModuleStyle['animate']
                : ModuleStyle['normal']
            }`}
          >
            Innovative Solutions{' '}
          </span>
          <span className={`block `}>
            for a{' '}
            <span
              className={` ${ModuleStyle['digitalWorldPart']} ${
                activeAnimationVariation == animationVariation[1]
                  ? ModuleStyle['animate']
                  : ModuleStyle['normal']
              }`}
            >
              Digital World
            </span>
          </span>
          <span
            className={`block  ${ModuleStyle['trustedAgencyPart']} ${
              activeAnimationVariation == animationVariation[2]
                ? ModuleStyle['animate']
                : ModuleStyle['normal']
            }`}
          >
            {' '}
            Trusted Agency
          </span>
        </p>
      </div>
      <div className={`mb-[5rem]`}>
        <p className={`text-center text-[#E9EAEC] text-lg`}>
          Elevate Your Brand: Powerhouse Agency for Impactful Strategies
        </p>
      </div>
      <div className={`flex justify-between space-x-[1.5rem]`}>
        <div>
          <Button.RedirectButton
            btnText="Our Service"
            colorSchema={BtnColorSchema.TransparentBgTextPrimary500}
            isArrow={false}
            isOpenNewTab={true}
            redirectLink=""
          />
        </div>
        <div>
          <Button.RedirectButton
            btnText="Work With Us"
            colorSchema={activeButtonVariant}
            isArrow={false}
            isOpenNewTab={true}
            redirectLink=""
          />
        </div>
        <div></div>
      </div>
    </div>
  );
};
