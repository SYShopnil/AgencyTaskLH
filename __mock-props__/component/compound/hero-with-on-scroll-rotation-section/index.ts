import { IHeroWithOnScrollRotation } from '@src/types/compound';
import { HeroSectionWithOnScrollRoationMockData } from '../../root/hero-section-with-on-scroll-rotaition';

export const hero_with_on_scroll_rotation_section_mockData: IHeroWithOnScrollRotation =
  {
    onScrollRotationSectionData: HeroSectionWithOnScrollRoationMockData,
  };
