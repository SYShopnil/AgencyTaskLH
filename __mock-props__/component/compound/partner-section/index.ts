import { IPartnerSection } from '@src/types/compound';
import { BannerImageMockData } from '../../root/banner-bar';

export const partnerSectionMockData: IPartnerSection = {
  banners: BannerImageMockData,
  heading: 'Trusted by Top-tier product campanies',
};
