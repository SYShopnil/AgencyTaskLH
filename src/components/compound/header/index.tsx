import { Navbar } from '@src/components/root';
import IconStore from '@src/components/root/_icon';
import { IHeader } from '@src/types/compound';
import { INavbar, IconEnums } from '@src/types/root';
import React from 'react';

export const Header = ({
  logo,
  navBarData: { commonPart, getInTouchLink, navItems },
  dataTestId,
}: IHeader) => {
  return (
    <div
      className={`grid grid-cols-12 gap-[5rem]`}
      data-testid={dataTestId || 'headerCatchy'}
    >
      <div className={`w-[6.837rem] h-[3.75rem] fill-primary_500 col-span-2`}>
        <IconStore iconName={logo} />
      </div>
      <div className={`col-span-10`}>
        <Navbar
          commonPart={commonPart}
          getInTouchLink={getInTouchLink}
          navItems={navItems}
        />
      </div>
    </div>
  );
};
