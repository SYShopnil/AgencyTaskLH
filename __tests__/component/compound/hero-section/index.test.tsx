import { HeroSection } from '@src/components/compound';
import { render } from '@testing-library/react';

describe('When Hero Section has been loaded', () => {
  it('Expect it will render successfully', () => {
    const { getByTestId } = render(<HeroSection />);
    const catchTheHeroSectionWrapper = getByTestId(`heroSectionCatcher`);
    expect(catchTheHeroSectionWrapper).toBeInTheDocument();
    expect(catchTheHeroSectionWrapper).toMatchSnapshot();
  });
});
