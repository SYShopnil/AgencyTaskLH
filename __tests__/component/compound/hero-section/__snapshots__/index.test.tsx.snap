// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Hero Section has been loaded Expect it will render successfully 1`] = `
<div
  class="flex justify-center items-center flex-col p-3"
  data-testid="heroSectionCatcher"
>
  <div
    class="w-[100%] h-[auto] mb-[2.5rem]"
  >
    <p
      class="text-center text-3xl  text-[#E9EAEC]"
    >
      <span
        class=" innovativeSolutionPart animate"
      >
        Innovative Solutions
         
      </span>
      <span
        class="block "
      >
        for a
         
        <span
          class=" digitalWorldPart normal"
        >
          Digital World
        </span>
      </span>
      <span
        class="block  trustedAgencyPart normal"
      >
         
        Trusted Agency
      </span>
    </p>
  </div>
  <div
    class="mb-[5rem]"
  >
    <p
      class="text-center text-[#E9EAEC] text-lg"
    >
      Elevate Your Brand: Powerhouse Agency for Impactful Strategies
    </p>
  </div>
  <div
    class="flex justify-between space-x-[1.5rem]"
  >
    <div>
      <a
        href="/"
        target="_blank"
      >
        <span
          class="flex justify-center items-center  "
        >
          <span
            class="text-base_2  flex justify-center items-center px-[1.5rem] py-[1rem] rounded-primary  bg-[transparent] hover:bg-primary_500 duration-[0.5s] text-primary_500 hover:text-neutral_500 border-[0.063rem] border-primary_500  w-[100%] h-[100%]"
          >
            <span>
              Our Service
            </span>
          </span>
        </span>
      </a>
    </div>
    <div>
      <a
        href="/"
        target="_blank"
      >
        <span
          class="flex justify-center items-center  "
        >
          <span
            class="text-base_2  flex justify-center items-center px-[1.5rem] py-[1rem] rounded-primary  bg-primary_500 hover:bg-[transparent]  duration-[0.5s] text-neutral_500 hover:text-primary_500  border-[0.063rem] border-primary_500  w-[100%] h-[100%]"
          >
            <span>
              Work With Us
            </span>
          </span>
        </span>
      </a>
    </div>
    <div />
  </div>
</div>
`;
